authentication/active_directory_authentication_provider.py

```diff
@@ -0,0 +1,53 @@
+#########
+# Copyright (c) 2016 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  * See the License for the specific language governing permissions and
+#  * limitations under the License.
+
+from ldap_authentication_provider import LDAPAuthenticationProvider
+
+DEFAULT_AUTH_ATTR_NAME = 'sAMAccountName'
+
+
+class ActiveDirectoryAuthenticationProvider(LDAPAuthenticationProvider):
+
+    def __init__(self, args, domain_name=None, **kwargs):
+        if 'search_props' in kwargs:
```

Comment:
sorry for the hassle, but this is user-facing, so we should rename to 'search_properties'